File: C/Empresa.c
typedef struct {
    char nome[100];
    int id;
    char cargo[100];
    float salario;
    int idDepartamento;
} Empregado;

typedef struct {
    char nome[100];
    int id;
    char localizacao[100];
} Departamento;

void atualizarInformacoesPessoais(Empregado *empregado, char novoNome[100], char novoCargo[100], float novoSalario);
void transferirDepartamento(Empregado *empregado, int novoIdDepartamento);
void obterInformacoesDepartamento(Empregado *empregado);

void adicionarEmpregadoDepartamento(Departamento *departamento, Empregado *empregado);
void removerEmpregadoDepartamento(Departamento *departamento, int idEmpregado);
void obterInformacoesEmpregadosDepartamento(Departamento *departamento);

File: haskell/Empresa.hs
data Empregado = Empregado {
    nome :: String,
    id :: Int,
    cargo :: String,
    salario :: Float,
    idDepartamento :: Int
} deriving (Show)

data Departamento = Departamento {
    nome :: String,
    id :: Int,
    localizacao :: String
} deriving (Show)

atualizarInformacoesPessoais :: Empregado -> String -> String -> Float -> Empregado
atualizarInformacoesPessoais empregado novoNome novoCargo novoSalario = empregado

transferirDepartamento :: Empregado -> Int -> Empregado
transferirDepartamento empregado novoIdDepartamento = empregado

obterInformacoesDepartamento :: Empregado -> IO ()
obterInformacoesDepartamento empregado = putStrLn ""

adicionarEmpregadoDepartamento :: Departamento -> Empregado -> Departamento
adicionarEmpregadoDepartamento departamento empregado = departamento

removerEmpregadoDepartamento :: Departamento -> Int -> Departamento
removerEmpregadoDepartamento departamento idEmpregado = departamento

obterInformacoesEmpregadosDepartamento :: Departamento -> IO ()
obterInformacoesEmpregadosDepartamento departamento = putStrLn ""
